core/src/forwarding_stage.rs

```diff
@@ -406,7 +412,9 @@ impl<VoteClient: ForwardingClient, NonVoteClient: ForwardingClient>
 
                     let mut batch = Vec::with_capacity(FORWARD_BATCH_SIZE);
                     core::mem::swap(&mut batch, &mut vote_batch);
-                    self.vote_client.send_transactions_in_batch(batch);
+                    if self.vote_client.send_transactions_in_batch(batch).is_err() {
```

Comment:
This logic looks transparent but unfortunately there is repetition of the code. I don't feel like it makes sense to extract it to some function or closure. Like:

```rust
let mut try_send_batch = |batch: &mut Vec<Vec<u8>>,
                          client: &dyn ForwardingClient,
                          forwarded_counter: &mut usize,
                          dropped_counter: &mut usize| {
    if batch.len() == FORWARD_BATCH_SIZE {
        *forwarded_counter += batch.len();

        let mut swap_batch = Vec::with_capacity(FORWARD_BATCH_SIZE);
        std::mem::swap(batch, &mut swap_batch);

        if client.send_transactions_in_batch(swap_batch).is_err() {
            *dropped_counter += FORWARD_BATCH_SIZE;
        }
    }
};
```

Reply:
I think a small function would be a good thing here, as it will reduce the fragility of the code during further refactors.

Reply:
done